src/gui/layertree/qgslayertreeviewdefaultactions.cpp

```diff
@@ -274,6 +292,15 @@ void QgsLayerTreeViewDefaultActions::zoomToLayer()
   QApplication::restoreOverrideCursor();
 }
 
+void QgsLayerTreeViewDefaultActions::zoomToSelection()
+{
+  QAction *s = qobject_cast<QAction *>( sender() );
+  QgsMapCanvas *canvas = reinterpret_cast<QgsMapCanvas *>( s->data().value<void *>() );
+  QApplication::setOverrideCursor( Qt::WaitCursor );
```

Comment:
Can you please change this to:

    QgsTemporaryCursorOverride cursorOverride( Qt::WaitCursor );
    zoomToSelection( canvas );

(I'd like to avoid all direct use of setOverrideCursor/restoreOverrideCursor in new code)